synapse/appservice/api.py

```diff
@@ -72,6 +72,28 @@ def query_alias(self, service, alias):
         defer.returnValue(False)
 
     @defer.inlineCallbacks
+    def query_3pu(self, service, protocol, fields):
+        uri = service.url + ("/3pu/%s" % urllib.quote(protocol))
```

Comment:
I copied it from other examples in the file, but yes I agree your line is neater.